Tests/SwiftParserTest/IncrementalParsingTests.swift

```diff
@@ -37,4 +37,395 @@ public class IncrementalParsingTests: XCTestCase {
+
+  public func testMultiEditMapping() throws {
+    try XCTSkipIf(true, "Swift parser does not handle node reuse yet")
+    assertIncrementalParse(
+      """
+      let one: Int;let two: Int; let three: Int; ⏩️⏸️                      ⏪️⏩️⏸️   ⏪️let found: Int;let five: Int;
+      """,
+      reusedNodes: [
+        ReusedNodeSpec("let one: Int;", kind: .codeBlockItem),
+        ReusedNodeSpec("let two: Int;", kind: .codeBlockItem),
+        ReusedNodeSpec("let five: Int;", kind: .codeBlockItem),
+      ]
+    )
+  }
+
+  public func testAddProperty() throws {
+    try XCTSkipIf(true, "Swift parser does not handle node reuse yet")
+    assertIncrementalParse(
+      """
+      struct Foo {
+        let a: Int
+        let b: Int
+        let c: Int
+        let d: String
+        ⏩️⏸️let e_newProp: String⏪️
+        let f: Int
+        let g: Int
+      }
+      """,
+      reusedNodes: [
+        ReusedNodeSpec("let a: Int", kind: .memberDeclListItem),
+        ReusedNodeSpec("let b: Int", kind: .memberDeclListItem),
+        ReusedNodeSpec("let c: Int", kind: .memberDeclListItem),
+        ReusedNodeSpec("let g: Int", kind: .memberDeclListItem),
+      ]
+    )
+  }
+
+  public func testWarpInClass() throws {
+    try XCTSkipIf(true, "Swift parser does not handle node reuse yet")
+    assertIncrementalParse(
+      """
+      ⏩️⏸️class Foo {⏪️
+        func foo1() {
+          print("Hello Foo!")
+        }
+
+        func foo2() {
+          print("Hello again")
+        }
+      """,
+      reusedNodes: [
+        ReusedNodeSpec(
+          """
+          func foo1() {
+            print("Hello Foo!")
+          }
+          """,
+          kind: .functionDecl
+        ),
+        ReusedNodeSpec(
+          """
+          func foo2() {
+            print("Hello again!")
+          }
+          """,
+          kind: .functionDecl
+        ),
+      ]
+    )
+  }
+
+  public func testUnwarpClass() throws {
+    try XCTSkipIf(true, "Swift parser does not handle node reuse yet")
+    assertIncrementalParse(
+      """
+      ⏩️class Bar {⏸️⏪️
+        func bar1() {
+          let pi = 3.1415
+          print("Pi is (approximately) \\(pi)")
+        }
+
+        func bar2() {
+          print("I can compute Pi as well:")
+          bar1()
+        }
+      }
+      """,
+      reusedNodes: [
+        ReusedNodeSpec(
+          """
+          func bar1() {
+            let pi = 3.1415
+            print("Pi is (approximately) \\(pi)")
+          }
+          """,
+          kind: .functionDecl
+        ),
+        ReusedNodeSpec(
+          """
+          func bar2() {
+            print("I can compute Pi as well:")
+            bar1()
+          }
+          """,
+          kind: .functionDecl
+        ),
+      ]
+    )
+  }
+
+  public func testNextTokenCalculation() throws {
+    try XCTSkipIf(true, "Swift parser does not handle node reuse yet")
+    assertIncrementalParse(
+      """
+      let a = "hello"
+          let c = "⏩️ ⏸️⏪️world"
```

Comment:
I don’t recall why but in reuse.swift it has the following comment and `let a` is indented by four spaces. I think we should maintain that indentation and comment.

```swift
// The indentation on these lines is important for the test case
```